GameEngine/Engine.cs

```diff
@@ -88,5 +87,28 @@ public static void Start(Func<IGameData> gameDataFactory, int loadFromSlot = 0)
                 }
             }
         }
+
+        public void StartNewGame()
+        {
+            string intro = gameData.GetGameIntroduction();
+            Console.Clear();
+            Console.WriteLine(intro);
+
+            Console.WriteLine($"First, we are going to set up your Character.");
+            Console.WriteLine($"Now, your name is, {gameData.PlayerName}, do you want to change it? Yes/No");
+
+            if(Console.ReadLine().Equals("yes", StringComparison.OrdinalIgnoreCase))
+            {
+                string playersNewName;
+                Console.Write($"You are changing your name, what is your new name?: ");
+                playersNewName = Console.ReadLine();
+                gameData.PlayerName = playersNewName;
```

Comment:
You're still assigning the player name into the game data class here. 
Let's make these extra changes:

In the interface and the data class change this property to be DefaultPlayerName instead of just PlayerName because it will only be the default if they don't choose their own.

In the interface write it as { get; } This will make it so the engine (which is only working with the interface) can only get the data and not set it. In the data class you can keep it as { get; set;}, but from the perspective of the engine, this set will not be visible. This will help enforce the business rule we've set up that this gamedata class should be read-only.

Can you make the IGameData interface so it doesn't allow setting this?